imcsdk/imcsession.py

```diff
@@ -415,36 +415,43 @@ def __validate_connection(self):
 
     def __validate_imc(self):
         """ ValidateIMC method validates if a given host is valid IMC Server."""
-        from .imcmethodfactory import config_resolve_class
+
+        from .mometa.top.TopSystem import TopSystem
+        from .mometa.compute.ComputeRackUnit import ComputeRackUnit
+        from .imcmethodfactory import config_resolve_class, config_resolve_dn
+
         valid_models = ("R460-4640810", "C260-BASE-2646")
         model_validated = False
+        is_imc = False
 
-        rack_elem = config_resolve_class(cookie=self.__cookie,
-                                         class_id="computeRackUnit")
-
+        rack_dn = ComputeRackUnit(TopSystem().rn, "1").dn
+        rack_elem = config_resolve_dn(cookie=self.__cookie, dn=rack_dn)
         rack_elem_response = self.post_elem(rack_elem)
 
         if rack_elem_response.error_code == 0:
-            rack = rack_elem_response.out_configs.child[0]
-            model_name = rack.model
-            if model_name.startswith("UCSC"):
-                model_validated = True
-            elif model_name.startswith("UCS-E"):
-                model_validated = True
-            elif model_name in valid_models:
-                model_validated = True
+            for rack in rack_elem_response.out_config.child:
+                model_name = rack.model
+                if model_name.startswith("UCSC"):
+                    model_validated = True
+                elif model_name.startswith("UCS-E"):
+                    model_validated = True
+                elif model_name in valid_models:
+                    model_validated = True
         if not model_validated:
             return False
 
-        # nw_elem = self.config_resolve_class(cookie=self.__cookie,
-        #                                class_id="networkElement")
-        #
-        # nw_elem_response = self.post_elem(nw_elem)
-        # # To Do - returns error tag
-        # if nw_elem_response.error_code == 0:
-        #     self.__clear()
-        #     return False
-
+        nw_elem = config_resolve_class(cookie=self.__cookie,
```

Comment:
@prvashis Can you take this block out into a new function and call it `is_fabric_interconnect`.